### PR TITLE
update and clean up README.md

### PR DESCRIPTION
**Pull request by mkernohanbc**
This change is a minor refresh of README.md to reflect current status of work on the design system:

- Adds links to new resources publishing alongside 3.0.0 release
- Updates status of DS sub-products
- Updates language around archiving of legacy design system resources
- General content cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,66 +1,59 @@
...
-* [Developer documentation](https://classic.developer.gov.bc.ca/Design-System/About-the-Design-System)
-* [Source code](https://github.com/bcgov/design-system/tree/main/components)
+- [Developer documentation](https://classic.developer.gov.bc.ca/Design-System/About-the-Design-System)
+- [Source code](https://github.com/bcgov/design-system/tree/main/components)
 
 ## License
 
     Copyright 2016 Province of British Columbia
 
-    Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (ty2k) [inline README.md:56]: Without these line breaks, this block overflows:

Screenshot 2024-04-23 at 12 58 30 PM

Maybe we just take this summary license section out entirely and replace it with "See LICENSE file"?

**Review by ty2k — CHANGES_REQUESTED** [review]
Content is all good, but there's two small formatting changes we should make:

File: README.md (hunk 1 above)
Comment (mkernohanbc) [reply README.md:56]: Weird! I think you're right — now that I think about it more closely, I don't know why this section is here at all. We have LICENSE.md accessible right here: 

Screenshot 2024-04-24 at 10 10 49

I've just made another commit (3f6742b) to remove that paragraph from README.md entirely.

File: README.md [hunk 2]
```diff
@@ -1,66 +1,59 @@
 # B.C. Design System [![img](https://img.shields.io/badge/Lifecycle-Maturing-007EC6)](https://github.com/bcgov/repomountie/blob/master/doc/lifecycle-badges.md)
 
-> [!NOTE]
-> **February 2024**: The B.C. Design System is currently in active development. If you have questions, [contact the design system team](#contact) or [open an issue](https://github.com/bcgov/design-system/issues).
+> [!NOTE] > **April 2024**: The B.C. Design System is currently in active development. If you have questions, [contact the design system team](#contact) or [open an issue](https://github.com/bcgov/design-system/issues).
```
[the remark this answers is not in the document]
Comment (mkernohanbc) [reply README.md]: This is fixed (I think?) in [04d6a4d](https://github.com/bcgov/design-system/pull/337/commits/04d6a4d9a8750215a8676094587e4eb1309937b1)

**Review by ty2k — APPROVED** [review]
Looks great!

**Pull request merged**